js/gravityforms-mfi-multifile.js

```diff
@@ -0,0 +1,290 @@
+                fieldID = up.settings.multipart_params.field_id,
+                totalCount = countFiles(fieldID),
+                disallowed = up.settings.gf_vars.disallowed_extensions,
+                extension;
+
+            if( max > 0 && totalCount >= max){
+                $.each(files, function(i, file) {
+                    up.removeFile(file);
+                    return;
+                });
+                return;
+            }
+            $.each(files, function(i, file) {
+
+                extension = file.name.split('.').pop();
+
+                if($.inArray(extension, disallowed) > -1){
+                    addMessage(up.settings.gf_vars.message_id, file.name + " - " + strings.illegal_extension);
+                    up.removeFile(file);
+                    return;
+                }
+
+                if ((file.status == plupload.FAILED) || (max > 0 && totalCount >= max)){
+                    up.removeFile(file);
+                    return;
+                }
+
+                var size = typeof file.size !== 'undefined' ? plupload.formatSize(file.size) : strings.in_progress;
+                var status = '<div id="'
+                    + file.id
+                    + '" class="ginput_preview">'
+                    + file.name
+                    + ' (' + size + ') <b></b> '
+                    + '<a href="javascript:void(0)" title="' + strings.cancel_upload + '" onclick=\'$this=jQuery(this); var uploader = gfMultiFileUploader.uploaders.' + up.settings.container + ';uploader.stop();uploader.removeFile(uploader.getFile("' + file.id +'"));$this.after("' + strings.cancelled + '"); uploader.start();$this.remove();\' onkeypress=\'$this=jQuery(this); var uploader = gfMultiFileUploader.uploaders.' + up.settings.container + ';uploader.stop();uploader.removeFile(uploader.getFile("' + file.id +'"));$this.after("' + strings.cancelled + '"); uploader.start();$this.remove();\'>' + strings.cancel + '</a>'
+                    + '</div>';
+
+                $('#' + up.settings.filelist).prepend(status);
+                totalCount++;
+
+            });
+
+            up.refresh(); // Reposition Flash
+
+            var formElementID = "form#gform_" + formID;
+            var uidElementID = "input:hidden[name='gform_unique_id']";
+            var uidSelector = formElementID + " " + uidElementID;
+            var $uid = $(uidSelector);
+            if($uid.length==0){
+                $uid = $(uidElementID);
+            }
+
+            uniqueID = $uid.val();
+            if('' === uniqueID){
+                uniqueID = generateUniqueID();
+                $uid.val(uniqueID);
+            }
+
+
+            if(max > 0 && totalCount >= max){
+                gfMultiFileUploader.toggleDisabled(up.settings, true);
+                addMessage(up.settings.gf_vars.message_id, strings.max_reached)
+            }
+
+
+            up.settings.multipart_params.gform_unique_id = uniqueID;
+            up.start();
```

Comment:
Since gravity's php code processing file uploads has no relevant hooks to allow modyfing the submitting form id, and that an upload referencing the random form id will fail, the upload process is strictly the same as without this plugin.
The multipart_param 'form_id' parameter is not modified, so the server receives the original form id, and processes the upload properly.